### PR TITLE
Allow to specify BUILD_ALIAS

### PR DESCRIPTION
**Pull request by apodhrad**
Allow to specify BUILD_ALIAS such as
  - CI
  - CR
  - Final

### CODE REVIEW

**Comment by apodhrad** [comment]
Now, you can also specify BUILD_TIMESTAMP (by default 'yyyyMMdd-HHmm').So, the version format looks like
{code}
2.0.0.SNAPSHOT-v20170526-1217
{code}

File: pom.xml [hunk 1]
```diff
@@ -124,6 +123,22 @@
...
+				<dependencies>
+					<dependency>
+						<groupId>org.eclipse.tycho.extras</groupId>
+						<artifactId>tycho-buildtimestamp-jgit</artifactId>
+						<version>${tychoExtrasVersion}</version>
+					</dependency>
+				</dependencies>
+				<configuration>
+					<strictBinIncludes>false</strictBinIncludes>
+					<format>'${BUILD_ALIAS}-v'${BUILD_TIMESTAMP}</format>
```
Comment (rawagner) [inline pom.xml]: why not ${BUILD_ALIAS}.v'${BUILD_TIMESTAMP}. Ie eclipse.core is using dot instead of dash

Comment (apodhrad) [inline pom.xml]: I used the jbosstools format since eclipse doesn't use the BUILD_ALIAS. Of course, we can discuss which format is best for our purpose.

**Comment by apodhrad** [comment]
Unfortunately, it is not possible to update from
```
2.0.0.SNAPSHOT-vXYZ
```
to
```
2.0.0.Final or 2.0.0.Final-vXYZ 
```

**Comment by apodhrad** [comment]
Changed the default build alias to 'CI' so that a user can update to the final bits.

**Comment by apodhrad** [comment]
Moreover, I think it is ok to have a timestamp for any BUILD_ALIAS.

File: pom.xml (hunk 1 above)
Comment (apodhrad) [inline pom.xml]: Moreover, I like the dash ;-)

**Pull request merged**